server/core/segment.go

```diff
@@ -0,0 +1,25 @@
+package core
+
+import (
+	"database/sql"
+	"log"
+)
+
+// TODO: normally we shouldn't export members of this domain object.
+type Segment struct {
+	Id   int64  `json:"id"`
+	Name string `json:"name"`
+}
+
+func (s *Segment) Save(db *sql.DB) error {
+	err := db.QueryRow(
```

Comment:
Use `QueryRowContext` instead. [Contexts are amazing](https://blog.golang.org/context) in go so use them wherever you can!

Reply:
[pgx ](https://github.com/jackc/pgx) is a pretty great sql driver. You might want to use that instead.

Reply:
Thanks :)